packages/superset-ui-demo/storybook/stories/plugins/plugin-chart-table/TableStories.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { withKnobs, number } from '@storybook/addon-knobs';
+import { SuperChart } from '@superset-ui/chart';
+import TableChartPlugin, { TableChartProps } from '@superset-ui/plugin-chart-table';
+import data, { birthNames } from './data';
+import { withResizableChartDemo } from '../../../shared/components/ResizableChartDemo';
+
+export default {
+  title: 'Legacy Chart Plugins|legacy-plugin-chart-table',
+  decorators: [withKnobs, withResizableChartDemo],
+};
+
+new TableChartPlugin().configure({ key: 'table' }).register();
+
+function paginated(props_: TableChartProps, pageSize = 50): TableChartProps {
+  const props: TableChartProps = { ...props_ };
+  if (props.formData) {
+    props.formData = {
+      ...props.formData,
+      page_length: pageSize,
+    };
+  }
+  // eslint-disable-next-line camelcase
+  if (props.queryData?.form_data) {
+    props.queryData.form_data = {
+      ...props.queryData.form_data,
+      page_length: pageSize,
+    };
+  }
+  return props;
+}
+
+function adjustNumCols(props: TableChartProps, numCols = 7): TableChartProps {
+  const newProps = { ...props };
+  if (props?.queryData.data) {
+    const { columns, records } = props.queryData.data;
+    const curSize = columns.length;
+    const newColumns = [...new Array(numCols)].map((_, i) => {
+      return columns[i % curSize];
+    });
+    newProps.queryData = {
+      ...props.queryData,
+      data: {
+        records,
+        columns: newColumns,
+      },
+    };
+  }
+  return newProps;
+}
+
+/**
+ * Load sample data for testing
+ * @param props the original props passed to SuperChart
+ * @param pageSize number of records perpage
+ * @param targetSize the target total number of records
+ */
+function loadData(props: TableChartProps, pageSize = 50, targetSize = 1042): TableChartProps {
+  if (!props.queryData) return props;
+  const dat = props.queryData?.data;
+  if (dat && dat.records.length > 0) {
+    while (dat.records.length < targetSize) {
```

Comment:
```ts
const { records } = data;
const { length } = records;
data.records = [...new Array(targetSize).keys()].map(i => records[i % length]);
```

Reply:
The time complexity becomes `O(n)` instead of `O(log n)`, but I guess we can live with in favor of cleaner code.

Reply:
this is also demo so I would say optional. can live with better perf too.